compiler/modules/front/codeGen.bal

```diff
@@ -183,17 +188,24 @@ function codeGenForeachStmt(CodeGenContext cx, bir:BasicBlock startBlock, Scope?
     bir:BasicBlock afterCondition = cx.createBasicBlock();
     bir:CondBranchInsn branch = { operand: condition, ifFalse: exit.label, ifTrue: afterCondition.label };
     loopHead.insns.push(branch);
-    cx.pushLoopContext(exit, loopStep);
+    cx.pushLoopContext(exit, ());
     Scope loopScope = { name: varName, reg: loopVar, prev: scope, isFinal: true };
     bir:BasicBlock? loopBody = check codeGenStmts(cx, afterCondition, loopScope, stmt.body);
-    cx.popLoopContext();
+    
+    bir:BasicBlock? loopStep = cx.loopContinueBlock();
     if !(loopBody is ()) {
-        bir:BranchInsn branchToLoopStep = { dest: loopStep.label };
+        loopStep = cx.loopContinueBlock() ?: cx.createBasicBlock();
```

Comment:
Why is this `cx.loopContinueBlock()` rather than `loopStep`? You assigned `loopStep` just above.